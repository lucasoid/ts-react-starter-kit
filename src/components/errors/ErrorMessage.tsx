import * as React from 'react';
import * as styles from './Error.css';

interface IErrorMessage {
    children?: any;
}
export const ErrorMessage: React.FC<IErrorMessage> = props => (
    <div className={styles.errorMessage}>{props.children}</div>
);
