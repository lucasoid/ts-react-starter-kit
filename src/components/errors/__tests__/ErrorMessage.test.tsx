import * as React from 'react';
import { shallow } from 'enzyme';
import { ErrorMessage } from '../ErrorMessage';
import { ThemeContext, Themes } from '~theme';

// mocking React's useContext. Ugly, but effective.
jest.mock('react', () => {
    const original = jest.requireActual('react');
    return {
        __esModule: true,
        ...original,
        useContext: jest.fn(),
    };
});

describe('ErrorMessage', () => {
    it('displays a themed error message', () => {
        // @ts-ignore  <-- tell the IDE to be cool about using mockImplementationOnce
        React.useContext.mockImplementationOnce(() => ({ theme: Themes.LIGHT }));
        let lightWrapper = shallow(<ErrorMessage>Oops</ErrorMessage>);
        expect(lightWrapper).toMatchSnapshot();

        // @ts-ignore
        React.useContext.mockImplementationOnce(() => ({ theme: Themes.DARK }));
        let darkWrapper = shallow(<ErrorMessage>Oops</ErrorMessage>);
        expect(darkWrapper).toMatchSnapshot();
    });
});
