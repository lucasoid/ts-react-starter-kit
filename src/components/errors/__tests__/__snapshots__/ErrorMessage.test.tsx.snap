// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage displays a themed error message 1`] = `
<div
  className="errorMessage light"
>
  Oops
</div>
`;

exports[`ErrorMessage displays a themed error message 2`] = `
<div
  className="errorMessage dark"
>
  Oops
</div>
`;
